src/index.js

```diff
@@ -4,15 +4,55 @@ import rangeParser from 'parse-numeric-range';
 import shiki from 'shiki';
 import sanitizeHtml from 'sanitize-html';
 
-let highlighter = null;
+// To make sure we only have one highlighter per theme in a process
+const highlighterCache = new Map();
+
+function getThemesFromSettings(settings) {
+  if (!settings.theme)
+    throw new Error(
+      'MDX Pretty Code: no theme was specified in shikiOptions.theme'
```

Comment:
May be a better experience to return a default string theme (and a warning) instead of throwing?

Reply:
returning `undefined` (which I think would resolve to "nord" by shiki) works

Reply:
@atomiks just realized the fallback isn't working.
would need to do
```
  if (!settings.theme) {
    return {default: undefined};
  }
```

Want me to do a fix PR?

Reply:
Sure! ...would be nice to have some tests 😄